site/src/components/context/MDXPageContext.ts

```diff
@@ -0,0 +1,18 @@
+import { createContext, Dispatch, SetStateAction } from "react";
+
+export type Heading = {
+  anchor: string;
+  element: HTMLHeadingElement;
+};
+
+type MDXPageContextProps = {
+  headings: Heading[];
+  setHeadings: Dispatch<SetStateAction<Heading[]>>;
+};
+
+const MDXPageContext = createContext<MDXPageContextProps>({
+  headings: [],
+  setHeadings: () => undefined,
+});
+
+export default MDXPageContext;
```

Comment:
How about co-locating React context with the hook and the provider? This will help us see the relationship between various parts of the same feature.

Here’s how I imagine this at the exterior:

```
import { ScrollSyncProvider, useScrollSyncAnchor } from "../../../shared/scroll-sync"`
```

```ts
const SomePage = () => {
  return (
   <HeadingScrollSyncProvider>
      <SomeContent />
   </HeadingScrollSyncProvider>
   );
}
```

```tsx
const SomeContent = () => {
  const a1Ref = useScrollSyncAnchor();
  const a2Ref = useScrollSyncAnchor();
  
  return <div>
  <h1><a href="#hello" ref={a1Ref}>⚓️</a> Hello</h1>
  <p>lorem ipsum</p>
  <h1><a href="#world" ref={a2Ref}>⚓️</a> World</h1>
  </div>
}
```

Reply:
Or even:

```ts
const ScrollSyncAnchor = /* useScrollSyncAnchor + <a>*/;
```

```ts
import { ScrollSyncAnchor } from "../../../shared/scroll-sync";

const StyledScollSyncAnchor = styled(ScrollSyncAnchor)`
  ...
`;

StyledScollSyncAnchor.defaultProps = {
  children: "⚓️"
}

const SomeContent = () => {
  return <div>
    <h1><ScrollSyncAnchor anchor="hello" /> Hello</h1>
    <p>lorem ipsum</p>
    <h2><ScrollSyncAnchor anchor="world" /> World</h2>
  </div>
}
```

Note that the anchor is separate from the header, like in 

- https://beta.reactjs.org
- https://www.typescriptlang.org/docs/handbook/typescript-from-scratch.html
- https://prettier.io/docs/en/options.html